ctapipe/instrument/camera.py

```diff
@@ -518,6 +533,69 @@ def get_shower_coordinates(self, x, y, psi):
 
         return longi, trans
 
+    def get_pixel_id(self, x, y):
+        '''
+        Return the camera pixel number which contains a given position (x,y) 
+        in the camera frame. The (x,y) coordinates can be arrays (of equal length),
+        for which the methods returns an array of pixel ids. A warning is raised if the 
+        position falls outside the camera.
+
+        Parameters
+        ----------
+        x: astropy.units.Quantity (distance) of horizontal position(s) in the camera frame
+        y: astropy.units.Quantity (distance) of vertical position(s) in the camera frame
+
+        Returns
+        -------
+        pixel_ids: Pixel number or array of pixel numbers. Returns -1 if position falls 
+                   outside camera
+        '''
+
+        if not np.all(self.pix_area == self.pix_area[0], axis=0):
```

Comment:
Use `if np.any(self.pix_area != self.pix_area[0])`. `all` always needs to test every value, `any` exits at the first `True` value it encounters.

Also this looks like it should be a `@lazyproperty`, like `all_pixel_areas_equal` or  better name you can come up with.

Reply:
also might want to use a tolerance, in case of any rounding error in case the area of the pixel is calculated somewhere. (or `np.isclose()` - is there an `anyclose()` function?),

Reply:
`np.any(~np.isclose(...`

Reply:
done in 704d7cc and 5c9c185. I marked the lazyproperty internal (leading underscore). Correct me if that's a wrong use-case.